tests/e2e/test_operators.py

```diff
@@ -437,3 +437,70 @@ def test_logical_operators(update_environment):
             )
             == 4
         ), "Testcase #6 failed"
+
+
+@pytest.mark.e2e
+@pytest.mark.parametrize(
+    "rulebook",
+    [
+        pytest.param(
+            "test_string_search_match.yml",
+            id="string_search_match",
+        ),
+        pytest.param(
+            "test_string_search_search.yml",
+            id="string_search_search",
+        ),
+        pytest.param(
+            "test_string_search_regex.yml",
+            id="string_search_regex",
+        ),
+    ],
+)
```

Comment:
Hi @ttuffin you are binding the assertion logic for three different rulebooks, If we need to modify or update some of these tests it would require a refactor splitting them instead of just adding or update the testcase, it doesn't make sense, for me is a bad use of the DRY principle. I think it should be three diferent tests or only one rulebook.

Reply:
@Alex-Izquierdo actually the idea is that they _should_ be bound. My reasoning was that they all perform a similar operation (string search) using a different approach (match, search regex). Therefore I wanted to ensure they are fed the same dataset and get to the same end result via their respective approaches. If a test is added/removed/modified for one then the same should be done with the other two. I tried one rulebook but it was becoming unmaintainable. It does mean having to update three different rulebooks, but this will at least be enforced when the test fails to ensure they remain consistent. Splitting to three different tests will mean we could lose that consistency and they will diverge. If we find later on that is what we want, then they can easily be split out at that time?

Reply:
`match` has different behavior than `regex` or `search`, it must be decoupled.

Reply:
Tests decoupled